### ISSUE TITLE
Remove obsolute if-statement with check if encoding is present

### ISSUE THREAD

**Issue by goessebr** [OnroerendErfgoed/skosprovider_atramhasis#120]
The coverage tests on the master branch are broken. 
![image](https://user-images.githubusercontent.com/1926276/146163490-22e39532-6ad9-4597-b013-6a2cbc257344.png)

The code in https://github.com/OnroerendErfgoed/skosprovider_atramhasis/blob/a4f976295fbae783dafdab31c56d7774dc4dc99c/skosprovider_atramhasis/providers.py#L266-L267 is probably obsolete. A response object seems to get a default encoding of utf-8 if no charset is given.
(noticed in https://github.com/OnroerendErfgoed/skosprovider_atramhasis/pull/119#issuecomment-994573213)

This code was probably obsolete before but coverall seems to have a problem with it now due the algoritm used to calculate the coverage.

The obsolete code can be removed, I'm not not sure about the test. I might be safe to keep it.

@koenedaele @Wim-De-Clercq your opinions?

**Comment by Wim-De-Clercq** [comment]
The coverage check is only broken for this 1 commit, next commit would be fine. This can be safely ignored.

As for the encoding, a JSON response does not receive a charset in the header because it is technically illegal for `application/json` to have one.

Setting the encoding should be unnecessary. See more at https://github.com/Pylons/pyramid/issues/2860 and the links there.
**tldr;** it would be illegal to encode a json response in something else than utf-8, 16 or 32. 

And the `requests` library knows this
```py
        if not self.encoding and self.content and len(self.content) > 3:
            # No encoding set. JSON RFC 4627 section 3 states we should expect
            # UTF-8, -16 or -32. Detect which one to use; If the detection or
            # decoding fails, fall back to `self.text` (using charset_normalizer to make
            # a best guess).
```
The encoding parameter we set would give a slight speedboost because requests doesn't have to go decide between utf 8, 16, or 32. But it's probably total overkill for us to do so.

I'm all for removing it.

**Comment by koenedaele** [comment]
Looking at that code, in other projects we retry requests a few times before giving up. Would it be useful to do the same here?

**Comment by Wim-De-Clercq** [comment]
Yea I don't see why not.
